nservicebus/upgrades/5to6/index.md

```diff
@@ -10,20 +10,189 @@ related:
 
 For larger solutions the Visual Studio extension [Target Framework Migrator](https://visualstudiogallery.msdn.microsoft.com/47bded90-80d8-42af-bc35-4736fdd8cd13) can reduce the manual effort required in performing an upgrade.
 
 
-## IBus, IStartableBus and the Bus Static class are now obsolete
+### Update NServiceBus dependencies
+
+All NServiceBus dependencies for an endpoint project are managed via NuGet. Open the Manage NuGet Packages window for the endpoint project, switch to the Updates tab and look for packages that start with NServiceBus. Update each one to the latest Version 6 package.
+
+(screenshot?)
+
+NOTE: All of the NuGet packages are currently available as prerelease builds so the Include prerelease option must be selected by either checking the box in the Manage NuGet Packages window or by including the `-Pre` flag in the Package Manager Console. 
+
+Once all of the NServiceBus packages have been updated to Version 6 the project will contain quite a few errors. This is expected as a lot of things have changed.
+
+
+### Update Endpoint configuration
+
+In previous versions of NServiceBus, to connect a process to the transport, an instance of `IBus` was needed. In Version 6 and above, this concept has been deprecated and now an instance of `IEndpointInstance` is required. The code required to create and configure an `IEndpointInstance` is very similar to the code found in Version 5 endpoints for creating and configuring `IBus` instances.
+
+NOTE: This section describes updating a self-hosted endpoint. For endpoints that rely on the NServiceBus Host, see: [NServiceBus Host Upgrade Version 6 to 7](host-6to7.md).
+
+First, change all mentions of `BusConfiguration` to `EndpointConfiguration`. Note that `EndpointConfiguration` has a required constructor parameter to set the endpoint name. In Version 5, the name of the endpoint was provided via the `.EndpointName(name)` method on the `BusConfiguration` class. This call is no longer required in Version 6 and the method has been deprecated.
+
+Most of the other method calls on `EndpointConfiguration` shoud continue to work the same way as they did on `BusConfiguration`. The methods that have changed between versions will each have deprecation messages that describe how to achieve the same effect in Version 6.
+
+Once the instance of `EndpointConfiguration` has been created, it can be used to create an `IEndpointInstance`. In Version 5 and below, this step is accomplished using the `Bus` static class. In Version 6, this has been replaced with an `Endpoint` static class that works in a similar manner. 
+
+In Version 6 and above, any operation that interacts with the transport is asynchronous and returns a `Task`. This includes the `Start` method on the static `Endpoint` class and the `Stop` method on `IEndpointInstance`. Ideally these methods are called from within an `async` method and the results can simply be `awaited` (with `ConfigureAwait(false)` applied to them).
+
+```CSharp
```

Comment:
I wanted to make sure that we wanted to go to this level of depth before pulling this out as a snippet. There are no other snippets on the page other than these two and they are replicated elsewhere. I will pull them in to snippets with the next round of edits.